File: arrays_and_strings/lookandsay.py
# look and say

def look_and_say():

    text = "1"
    times = 5

    while times > 0:
        start = 0
        count = 1
        res = ""

        if len(text) == 1:
            res = "11"

        for i in range(1, len(text)):
            if text[i] == text[i-1]:
                count += 1
            else:
                res += str(count) + str(text[i-1]) 
                count = 1

        res += str(count) + str(text[i-1])
        
        print(res)
        text = res
        times -= 1

look_and_say()
        

        

File: arrays_and_strings/look_and_say.php
<?php

/*
look and say
*/

class Solution {

    public function look_and_say($string, $i, $loops) {

        // base case
        if ($i == $loops) {
            return $string;
        }

        $string .= " "; // adding dummy space to capture last sequence
        $repeat = $string[0];
        $count = 1;
        $output = "";

        for ($j = 1; $j < strlen($string); $j++) {
            if ($string[$j] == $repeat) {
                $count += 1;
            } else {
                $output .= $count . $repeat;
                $repeat = $string[$j];
                $count = 1;
            }
        }

        $i += 1;

        return $this->look_and_say($output, $i, $loops);
    }
}

$s = new Solution();
$loops = 10;
$ans = $s->look_and_say("1", 0, $loops);

echo $ans;

?>